blockly-rtc/src/websocket_handlers.js

```diff
@@ -33,10 +48,29 @@ export async function getEvents(serverId) {
   });
 };
 
+/**
+ * Add an entry to the database.
+ * @param {!LocalEntry} entry The entry to be added to the database.
+ * @return {!Promise} Promise object that represents the success of the write.
+ * @public
+ */
 export async function writeEvents(entry) {
   return new Promise((resolve, reject) => {
-    socket.emit('addEvents', entry, (serverId) => {
-      resolve(serverId);
+    socket.emit('addEvents', entry, () => {
+      resolve();
     });
   });
 };
+
+/**
+ * Listen for events broadcast by the server.
+ * @param {!Function} callback The callback handler that passes the events to
+ * the workspace.
+ * @return {!Promise} Promise object that represents the success of the write.
+ * @public
+ */
+export function getBroadcast(callback) {
+  socket.on('broadcastEvents', (rows)=> {
```

Comment:
Naming question: everywhere else you're calling it your variable `entry`.  Here you're calling the variable `rows`.  Why?

Reply:
Good catch. It used to be `rows` when each Blockly Event was written to the database in a separate row. Now, Blockly Events are grouped and added to the database as one row, or `entry`.

I go through looking for any places where it is still called `rows` and change it to `entry` in a following commit to this PR.